src/apps/KintoAppRegistry.sol

```diff
@@ -443,12 +395,15 @@ contract KintoAppRegistry is
 
         // Sets Child to parent contract
         for (uint256 i = 0; i < appContracts.length; i++) {
-            if (walletFactory.walletTs(appContracts[i]) > 0) revert CannotRegisterWallet();
+            address appContract = appContracts[i];
+            if (walletFactory.walletTs(appContract) > 0) revert CannotRegisterWallet(appContract);
             if (
-                childToParentContract[appContracts[i]] != address(0)
-                    && childToParentContract[appContracts[i]] != parentContract
-            ) revert ChildAlreadyRegistered();
-            childToParentContract[appContracts[i]] = parentContract;
+                childToParentContract[appContract] != address(0) && childToParentContract[appContract] != parentContract
```

Comment:
this check should also be done for the parent contract. Parent contract cannot be one of these

Reply:
Done.